package/cpp/api/JsiSkImage.h

```diff
@@ -70,20 +76,38 @@ class JsiSkImage : public JsiSkWrappingSkPtrHostObject<SkImage> {
         count >= 1 ? static_cast<SkEncodedImageFormat>(arguments[0].asNumber())
                    : SkEncodedImageFormat::kPNG;
 
-    auto quality = count == 2 ? arguments[1].asNumber() : 100.0;
+    auto quality = (count >= 2 && arguments[1].isNumber())
+                       ? arguments[1].asNumber()
+                       : 100.0;
     auto image = getObject();
     if (image->isTextureBacked()) {
       image = image->makeNonTextureImage();
     }
     sk_sp<SkData> data;
+
     if (format == SkEncodedImageFormat::kJPEG) {
       SkJpegEncoder::Options options;
       options.fQuality = quality;
       data = SkJpegEncoder::Encode(nullptr, image.get(), options);
+    } else if (format == SkEncodedImageFormat::kWEBP) {
```

Comment:
Is it possible to follow the logic from: https://github.com/google/skia/blob/main/modules/canvaskit/canvaskit_bindings.cpp#L947